src/ResourceManager/KeyVault/Commands.KeyVault/Commands/BackupAzureKeyVaultKey.cs

```diff
@@ -12,92 +12,129 @@
+                    ParameterSetName = ByKeyNameParameterSet,
+                    HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment.")]
         [ValidateNotNullOrEmpty]        
         public string VaultName { get; set; }
 
         /// <summary>
+        /// KeyBundle object to be backed up.
+        /// </summary>
+        /// <remarks>
+        /// Note that the backup applies to the entire family of a key (current and all its versions); 
+        /// since a key bundle represents a single version, the intent of this parameter is to allow pipelining.
+        /// The backup cmdlet will use the Name and VaultName properties of the KeyBundle parameter.
+        /// </remarks>
+        [Parameter( Mandatory = true,
+                    Position = 0,
+                    ValueFromPipelineByPropertyName = true,
+                    ParameterSetName = ByKeyObjectParameterSet,
+                    HelpMessage = "Key bundle to back up, pipelined in from the output of a retrieval call." )]
+        public KeyBundle Key { get; set; }
+
+        /// <summary>
         /// Key name
         /// </summary>
-        [Parameter(Mandatory = true,
-                   Position = 1,
-                   ValueFromPipelineByPropertyName = true,
-                   HelpMessage = "Key name. Cmdlet constructs the FQDN of a key from vault name, currently selected environment and key name.")]
+        [Parameter( Mandatory = true,
+                    Position = 1,
+                    ValueFromPipelineByPropertyName = true,
+                    ParameterSetName = ByKeyNameParameterSet,
+                    HelpMessage = "Key name. Cmdlet constructs the FQDN of a key from vault name, currently selected environment and key name.")]
         [ValidateNotNullOrEmpty]        
         [Alias(Constants.KeyName)]
         public string Name { get; set; }
 
         /// <summary>
         /// The output file in which the backup blob is to be stored
         /// </summary>
-        [Parameter(Mandatory = false,
-                   Position = 2,
-                   ValueFromPipelineByPropertyName = true,
-                   HelpMessage = "Output file. The output file to store the backed up key blob in. If not present, a default filename is chosen.")]
+        [Parameter( Mandatory = false,
+                    Position = 2,
+                    ValueFromPipelineByPropertyName = true,
+                    HelpMessage = "Output file. The output file to store the backed up key blob in. If not present, a default filename is chosen.")]
         [ValidateNotNullOrEmpty]
         public string OutputFile { get; set; }
 
+        /// <summary>
+        /// Instructs the cmdlet to overwrite the destination file, if it exists.
+        /// </summary>
+        [Parameter( Mandatory = false,
+                    Position = 3,
+                    ValueFromPipelineByPropertyName = true,
```

Comment:
I was following the existing pattern. It looks like you're ok with the current shape of these changes, and so I'll address this with the next opportunity. I have another change in the .. pipeline (no pun intended) for the 5/10 release. Thanks.